### PR TITLE
docs: Update release notes for recent 2.9 releases

### LINKED ISSUE #12330: Docs feedback: /docs/sources/release-notes/v2-9.md need update
and [changelog](https://github.com/grafana/loki/blob/v2.9.4/CHANGELOG.md) need update

### PR DESCRIPTION
**Pull request by JStickler**
**What this PR does / why we need it**:
We've had a couple of releases without updates to the release notes.

**Which issue(s) this PR fixes**:
Fixes #12330

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
missing v2.9.6? 🤔

File: docs/sources/release-notes/v2-9.md [hunk 1]
```diff
@@ -32,14 +32,37 @@ Grafana Labs is excited to announce the release of Loki 2.9.0 Here's a summary o
   - The `-ingester.unordered-writes` CLI flag is deprecated and will always default to `true` in the next major release.
   - For the full list of deprecations, see CHANGELOG.md
 
-
 ## Bug fixes
 
+### 2.9.5 (2024-02-28)
+
+* Action repo reference ([#11807](https://github.com/grafana/loki/issues/11807)) ([fb02797](https://github.com/grafana/loki/commit/fb02797a355e0d264d4cbd46045bd3ff2ab121d7)).
+* Bump base images and Go dependencies to address CVEs ([#12092](https://github.com/grafana/loki/issues/12092)) ([eee3598](https://github.com/grafana/loki/commit/eee35983f38fe04543b169ffa8ece76c23c4217b)).
```
Comment (DylanGuedes) [inline docs/sources/release-notes/v2-9.md:40]: those two are being listed as bug fixes, but they don't seem to be :thinking:

Comment (JStickler) [reply docs/sources/release-notes/v2-9.md:40]: I'll move them.  I threw this together really quickly at the end of the day.

**Comment by JStickler** [comment]
@zhangguanzhang 2.9.6 hasn't been released yet, it's still listed as Draft.

**Pull request merged**